datalad_dataverse/dataset.py

```diff
@@ -226,20 +221,57 @@ def rename_file(self,
             'pid': self._dsid,
         })
 
-        proc = self._api.update_datafile_metadata(
+        response = self.update_file_metadata(
             rename_id,
             json_str=datafile.json(),
             is_filepid=False,
         )
-        if proc.returncode:
-            raise RuntimeError(f"Renaming failed: {proc.stderr}")
+        response.raise_for_status()
 
         # https://github.com/datalad/datalad-dataverse/issues/236
         # we have no record to update the internal file list,
         # we must wipe it out
         self._files_latest = None
         self._dataset_latest = None
 
+    def update_file_metadata(self,
+                             identifier,
+                             json_str=None,
+                             is_filepid=False):
+
+        base_str = self._api.base_url_api_native
+        if is_filepid:
+            query_str = "{0}/files/:persistentId/metadata?persistentId={1}".format(
+                base_str, identifier
+            )
+        else:
+            query_str = "{0}/files/{1}/metadata".format(base_str, identifier)
+
+        headers = {}
+        if self._api.api_token:
+            headers["X-Dataverse-key"] = self._api.api_token
+
+        try:
+            resp = post(
+                query_str,
+                files={'jsonData': (None, json_str.encode())},
+                headers=headers
+            )
+            if resp.status_code == 401:
+                error_msg = resp.json()["message"]
+                raise ApiAuthorizationError(
+                    "ERROR: POST HTTP 401 - Authorization error {0}. MSG: {1}".format(
+                        query_str, error_msg
+                    )
+                )
+            return resp
+        except ConnectionError:
+            raise ConnectionError(
```

Comment:
This kills the chain of causes. For example: If there was some "Couldn't resolve host" or alike, we would not be able to find that anywhere in the error reported by the special remote.
Hence: `raise from`.